### PR TITLE
[Bugfix]: Nested components with text content cause error

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
I did some refactoring a bit, too -- to separate out fixture tests from manual test so that we can focus on individual problems, rather than rely on on big markdown files -- with node --inspect-brk and jest test name matching, it also means we'll be able to debug easier, too

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
image

**Comment by NullVoxPopuli** [comment]
Answer: nope. This was much easier than expected

File: index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const parseHBS = (node) => {
     return;
   }
   if (
-    /^({{|<[A-Z]|<\/[A-Z]|<[a-zA-Z]+\.[a-zA-Z])/g.test(node.value.trimStart())
```
Comment (NullVoxPopuli) [inline index.js:7]: removing this `^` was the fix

File: tests/index.test.js [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
+test('Nested components with nested components', () => {
+  assertTransform({
+    input: stripIndent`
+      <Component::With::Nesting>
+        <Nested::Component @foo="foo" @bar="bar" />
+        <Nested::Component @foo="foo" @bar="bar" />
+      </Component::With::Nesting>
+    `,
+    expected: stripIndent`
+      <Component::With::Nesting>
```
Comment (NullVoxPopuli) [inline tests/index.test.js:29]: there are some silly things happening with invisible character preservation / adding / removing

File: index.js (hunk 1 above)
Comment (josemarluedke) [reply index.js:7]: Just pushed a different approach to fix this issue without removing the `^`.

**Comment by josemarluedke** [comment]
Published in v0.3.3.

**Pull request merged**